File: CS/MyNavBar.cs
using System.Windows;
using DevExpress.Wpf.NavBar;

namespace MVVM_SelectedItem {
    public class MyNavBarControl : NavBarControl {
        public static readonly DependencyProperty BindableSelectedItemProperty;

        static MyNavBarControl() {
            BindableSelectedItemProperty = DependencyProperty.Register("BindableSelectedItem", typeof(object),
                typeof(MyNavBarControl), new PropertyMetadata(null, OnBindableSelectedItemChanged));
        }

        static void OnBindableSelectedItemChanged(DependencyObject d, DependencyPropertyChangedEventArgs e) {
            NavBarControl navBar = d as NavBarControl;
            foreach (NavBarGroup group in navBar.Groups)
                foreach (object item in group.Items) {
                    if (item == e.NewValue) {
                        group.SelectedItemIndex = group.Items.IndexOf(item);
                        return;
                    }
                }
        }

        public object BindableSelectedItem {
            get { return (object)GetValue(BindableSelectedItemProperty); }
            set { SetValue(BindableSelectedItemProperty, value); }
        }

        protected override void OnPropertyChanged(DependencyPropertyChangedEventArgs e) {
            base.OnPropertyChanged(e);
            if (e.Property == NavBarControl.SelectedItemProperty) {
                BindableSelectedItem = ((NavBarItem)e.NewValue).Content;
            }
        }
    }
}

File: VB/MyNavBar.vb
Imports Microsoft.VisualBasic
Imports System.Windows
Imports DevExpress.Wpf.NavBar

Namespace MVVM_SelectedItem
	Public Class MyNavBarControl
		Inherits NavBarControl
		Public Shared ReadOnly BindableSelectedItemProperty As DependencyProperty

		Shared Sub New()
			BindableSelectedItemProperty = _
			DependencyProperty.Register("BindableSelectedItem", GetType(Object), GetType(MyNavBarControl), _
			New PropertyMetadata(Nothing, AddressOf OnBindableSelectedItemChanged))
		End Sub

		Private Shared Sub OnBindableSelectedItemChanged(ByVal d As DependencyObject, _
			ByVal e As DependencyPropertyChangedEventArgs)

			Dim navBar As NavBarControl = TryCast(d, NavBarControl)
			For Each group As NavBarGroup In navBar.Groups
				For Each item As Object In group.Items
					If item Is e.NewValue Then
						group.SelectedItemIndex = group.Items.IndexOf(item)
						Return
					End If
				Next item
			Next group
		End Sub

		Public Property BindableSelectedItem() As Object
			Get
				Return CObj(GetValue(BindableSelectedItemProperty))
			End Get
			Set(ByVal value As Object)
				SetValue(BindableSelectedItemProperty, value)
			End Set
		End Property

		Protected Overrides Sub OnPropertyChanged(ByVal e As DependencyPropertyChangedEventArgs)
			MyBase.OnPropertyChanged(e)
			If e.Property Is NavBarControl.SelectedItemProperty Then
				BindableSelectedItem = (CType(e.NewValue, NavBarItem)).Content
			End If
		End Sub
	End Class
End Namespace